iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/gcp/HighPrivilegedRolesOnWorkloadResourcesCheck.java

```diff
@@ -19,49 +19,46 @@
  */
 package org.sonar.iac.terraform.checks.gcp;
 
-import org.sonar.check.Rule;
-import org.sonar.iac.terraform.api.tree.ExpressionTree;
-import org.sonar.iac.terraform.checks.ResourceVisitor;
-
 import java.util.List;
-import java.util.function.Predicate;
-import java.util.regex.Pattern;
-
-import static org.sonar.iac.terraform.checks.utils.PredicateUtils.containsMatchStringPredicate;
-import static org.sonar.iac.terraform.checks.utils.PredicateUtils.exactMatchStringPredicate;
-import static org.sonar.iac.terraform.checks.utils.PredicateUtils.treePredicate;
+import org.sonar.check.Rule;
+import org.sonar.iac.terraform.checks.AbstractNewResourceCheck;
 
 @Rule(key = "S6400")
-public class HighPrivilegedRolesOnWorkloadResourcesCheck extends ResourceVisitor {
+public class HighPrivilegedRolesOnWorkloadResourcesCheck extends AbstractNewResourceCheck {
 
   private static final String MESSAGE_FOR_BINDING = "Make sure it is safe to give those members full access to the resource.";
   private static final String MESSAGE_FOR_MEMBER = "Make sure it is safe to grant that member full access to the resource.";
   private static final String MESSAGE_ON_GRANT_FULL_ACCESS = "Make sure it is safe to grant full access to the resource.";
 
-  private static final Predicate<ExpressionTree> IS_OWNER_OR_MANAGER = treePredicate(exactMatchStringPredicate("MANAGER|OWNER", Pattern.CASE_INSENSITIVE));
-  private static final Predicate<ExpressionTree> CONTAINS_PRIVILEGED_ROLE = treePredicate(containsMatchStringPredicate("ADMIN|MANAGER|OWNER|SUPERUSER", Pattern.CASE_INSENSITIVE));
-
   @Override
   protected void registerResourceConsumer() {
     register(List.of(IAM_BINDING_RESOURCE_NAMES),
       resource -> resource.attribute("role")
-        .reportIf(CONTAINS_PRIVILEGED_ROLE, MESSAGE_FOR_BINDING));
+                          .reportIf(matchesPattern(".*(?:ADMIN|MANAGER|OWNER|SUPERUSER).*"), MESSAGE_FOR_BINDING));
```

Comment:
I'm not happy with the additional spaces here. The alignment of the "attribute()" and "reportIf()" methods means to me that they both apply to the resource symbol. However, we should find a common format/agreement. Otherwise, we will have a mixed style.

Reply:
I agree with your argument (and I also remember that you have used that argument once, and I agreed to it back then)